addons/l10n_it_edi_withholding/models/account_move.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from collections import namedtuple
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_amount_vat_signed = fields.Monetary(string='VAT', compute='_compute_amount_extended', readonly=True)
+    l10n_it_amount_pension_fund_signed = fields.Monetary(string='Pension Fund', compute='_compute_amount_extended', readonly=True)
+    l10n_it_amount_withholding_signed = fields.Monetary(string='Withholding', compute='_compute_amount_extended', readonly=True)
+    l10n_it_amount_before_withholding_signed = fields.Monetary(string='Before Withholding', compute='_compute_amount_extended', readonly=True)
+
+    @api.depends('amount_total_signed')
+    def _compute_amount_extended(self):
+        for move in self:
+            totals = dict(vat=0.0, withholding=0.0, pension_fund=0.0)
+            for line in [line for line in move.line_ids if move.is_invoice(True) and line.tax_line_id]:
+                totals[line.tax_line_id._l10n_it_get_tax_kind()] -= line.balance
+            move.l10n_it_amount_vat_signed = totals['vat']
+            move.l10n_it_amount_withholding_signed = totals['withholding']
+            move.l10n_it_amount_pension_fund_signed = totals['pension_fund']
+            move.l10n_it_amount_before_withholding_signed = move.amount_untaxed_signed + totals['vat'] + totals['pension_fund']
+
+    def _l10n_it_edi_prepare_fatturapa_tax_details(self, tax_details, reverse_charge_refund=False):
+        tax_details['tax_details'] = {k: v for k, v in tax_details['tax_details'].items() if v['tax']._l10n_it_filter_kind('vat')}
+        return super()._l10n_it_edi_prepare_fatturapa_tax_details(tax_details, reverse_charge_refund)
+
+    def _l10n_it_edi_filter_fatturapa_tax_details(self, line, tax_values):
+        """Filters tax details to only include the positive amounted lines regarding VAT taxes."""
+        return tax_values['tax_repartition_line'].factor_percent >= 0 and line['taxes']._l10n_it_filter_kind('vat').amount >= 0
```

Comment:
Should not we filter the tax details on vat taxes?  Maybe the tax details are not needed to be calculated for the withholding as you only do the totals per invoice.  (maybe we can avoid the method above this way)

Reply:
I don't understand. Tomorrow you tell me 😄

Reply:
Yes, by fixing the main tax_details function like you said (filtering all the tax_repartition lines that are not VAT), I also removed this function and tests still pass. 👍🏻